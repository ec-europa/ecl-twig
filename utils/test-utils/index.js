const Twig = require('twig');
const { format } = require('prettier');

const renderTwigFile = (file, options, cb) =>
  Twig.renderFile(file, options, (err, html) =>
    cb(err, format(html, { parser: 'html' }))
  );

const renderTwigFileAsNode = (file, options) =>
  new Promise((resolve, reject) =>
    Twig.renderFile(file, options, (err, html) => {
      if (err) return reject(err);

      const div = document.createElement('div');
      div.innerHTML = html.trim();

      return resolve(div.firstChild);
    })
  );

module.exports = {
  renderTwigFile,
  renderTwigFileAsNode,
};
