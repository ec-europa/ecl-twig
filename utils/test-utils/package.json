{
  "name": "@ecl-twig/test-utils",
  "version": "2.29.0",
  "description": "A local test helper",
  "main": "index.js",
  "private": true,
  "devDependencies": {
    "deepmerge": "4.2.2"
  }
}
