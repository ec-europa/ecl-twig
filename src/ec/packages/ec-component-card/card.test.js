import path from 'path';
import { merge, renderTwigFileAsNode } from '@ecl-twig/test-utils';

describe('EC - Card', () => {
  const template = path.resolve(__dirname, './card.html.twig');
  const render = params => renderTwigFileAsNode(template, params);
  const defaultIconPath = 'static/icons.svg';

  describe('Default', () => {
    const data = {
      card: {
        title: {
          type: 'standalone',
          path: '/example',
          label: 'Better regulation',
        },
        description:
          'Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.',
        meta: ['Meta 1', 'Meta 2', 'Meta 3'],
        image: {
          src:
            'https://inno-ecl.s3.amazonaws.com/media/examples/example-image.jpg',
          alt: 'Better regulation',
        },
        infos: [
          {
            label: '2018/10/22',
            icon: {
              type: 'general',
              name: 'calendar',
              path: defaultIconPath,
            },
          },
          {
            label: 'Luxembourg',
            icon: {
              type: 'general',
              name: 'location',
              path: defaultIconPath,
            },
          },
        ],
        tags: [
          {
            label: 'Tag 1',
            path: '/example-1',
          },
          {
            label: 'Tag 2',
            path: '/example-2',
          },
          {
            label: 'Tag 3',
            path: '/example-3',
          },
        ],
      },
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });

  describe('Tile', () => {
    const data = {
      card: {
        description:
          'Transparently designing and evaluating evidence-based EU legislation, backed by citizens views.',
        title: {
          label: 'Better regulation',
        },
        type: 'tile',
        links: [
          {
            label: 'link 1',
            path: '/example-1',
          },
          {
            label: 'link 2',
            path: '/example-2',
          },
          {
            label: 'link 3',
            path: '/example-3',
          },
        ],
      },
    };

    test('renders correctly', () => {
      expect.assertions(1);
      return expect(render(data)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(data, {
        extra_classes: 'custom-class custom-class--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(data, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });
  });
});
