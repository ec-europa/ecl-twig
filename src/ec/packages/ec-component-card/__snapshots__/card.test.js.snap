// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EC - Card Card default - renders correctly 1`] = `
"<article class=\\"ecl-card\\">
  <header class=\\"ecl-card__header\\">
    <div
      class=\\"ecl-card__image\\"
      alt=\\"card image\\"
      style=\\"background-image:url('https://v2--europa-component-library.netlify.com/example-image.jpg')\\"
    ></div>
    <div class=\\"ecl-card__meta\\">Meta 1 | Meta 2 | Meta 3</div>
    <h1 class=\\"ecl-card__title\\">
      <a href=\\"/example\\" class=\\"ecl-link ecl-link--standalone\\"
        >Better regulation</a
      >
    </h1>
  </header>
  <section class=\\"ecl-card__body\\">
    <div class=\\"ecl-card__description\\">
      Transparently designing and evaluating evidence-based EU legislation,
      backed by citizens views.
    </div>
  </section>
  <footer class=\\"ecl-card__footer\\">
    <ul class=\\"ecl-card__info-container\\">
      <li class=\\"ecl-card__info-item\\">
        <svg class=\\"ecl-icon ecl-icon--xs\\" focusable=\\"false\\" aria-hidden=\\"true\\">
          <use xlink:href=\\"static/icons.svg#general--calendar\\"></use>
        </svg>
        <span class=\\"ecl-card__info-label\\">2018/10/22</span>
      </li>
      <li class=\\"ecl-card__info-item\\">
        <svg class=\\"ecl-icon ecl-icon--xs\\" focusable=\\"false\\" aria-hidden=\\"true\\">
          <use xlink:href=\\"static/icons.svg#general--location\\"></use>
        </svg>
        <span class=\\"ecl-card__info-label\\">Luxembourg</span>
      </li>
    </ul>

    <ul class=\\"ecl-card__tag-container\\">
      <li class=\\"ecl-card__tag-item\\">
        <a href=\\"/example-1\\" class=\\"ecl-tag ecl-card__tag\\">
          Tag 1
        </a>
      </li>
      <li class=\\"ecl-card__tag-item\\">
        <a href=\\"/example-2\\" class=\\"ecl-tag ecl-card__tag\\">
          Tag 2
        </a>
      </li>
      <li class=\\"ecl-card__tag-item\\">
        <a href=\\"/example-3\\" class=\\"ecl-tag ecl-card__tag\\">
          Tag 3
        </a>
      </li>
    </ul>
  </footer>
</article>
"
`;

exports[`EC - Card Card tile - renders correctly 1`] = `
"<article class=\\"ecl-card ecl-card--tile\\">
  <header class=\\"ecl-card__header\\">
    <h1 class=\\"ecl-card__title\\">
      Better regulation
    </h1>
  </header>
  <section class=\\"ecl-card__body\\">
    <div class=\\"ecl-card__description\\">
      Transparently designing and evaluating evidence-based EU legislation,
      backed by citizens views.
    </div>
    <ul class=\\"ecl-card__link-container\\">
      <li class=\\"ecl-card__link-item\\">
        <a
          href=\\"/example-1\\"
          class=\\"ecl-link ecl-link--standalone ecl-card__link\\"
          >link 1</a
        >
      </li>
      <li class=\\"ecl-card__link-item\\">
        <a
          href=\\"/example-2\\"
          class=\\"ecl-link ecl-link--standalone ecl-card__link\\"
          >link 2</a
        >
      </li>
      <li class=\\"ecl-card__link-item\\">
        <a
          href=\\"/example-3\\"
          class=\\"ecl-link ecl-link--standalone ecl-card__link\\"
          >link 3</a
        >
      </li>
    </ul>
  </section>
  <footer class=\\"ecl-card__footer\\"></footer>
</article>
"
`;
